File: backend/models/coffeemaker.go
package models

type CoffeeMakerType string

const (
	EspressoMaker  = "espresso_maker"
	PouroverMaker  = "pourover_maker"
	ImmersionMaker = "immersion_maker"
	DripMaker      = "drip_maker"
	Other          = "other_maker"
)

type CoffeeMaker struct {
	model
	Name        string `gorm:"not null;default:null;unique" json:"name"`
	Description *string `json:"description"`
	Type        *CoffeeMakerType `json:"type"`
}

File: mobile/src/types/CoffeeMaker.ts
import { Model } from "./Model";

export enum CoffeeMakerType {
  EspressoMaker = "espresso_maker",
  PouroverMaker = "pourover_maker",
  ImmersionMaker = "immersion_maker",
  DripMaker = "drip_maker",
  Other = "other_maker",
}

export type CreateCoffeeMaker = {
  name: string;
  description?: string;
  type?: CoffeeMakerType;
};

export type CoffeeMaker = Model &
  CreateCoffeeMaker & {
    id: number;
  };
